src/tcgwars/logic/impl/gen4/GreatEncounters.groovy

```diff
@@ -2960,34 +2954,34 @@ public enum GreatEncounters implements LogicCardInfo {
             attackRequirement {}
             onAttack {
               damage 40
-              apply ASLEEP
-              def pcs = defending
-              targeted(pcs) {
-                if (pcs.isSPC(ASLEEP)) {// Is !bg.em().run(new ApplySpecialCondition(POISONED, pcs, SOURCE.ATTACK)) better here
-                  delayed {
-                    before ASLEEP_SPC, null, null, BEGIN_TURN, {
-                      flip "Asleep (Endless Darkness)", 2, {}, {}, [2: {
-                        ef.unregisterItself(bg.em());
-                      }, 1:{
-                        bc "$ef.target is still asleep."
-                      }, 0:{
-                        bc "$pcs is knocked out by $thisMove."
-                        new Knockout(pcs).run(bg)
-                      }]
-                      prevent()
-                    }
-                    after CLEAR_SPECIAL_CONDITION, pcs, {
-                      if (ef.types.contains(ASLEEP)) {
-                        unregister()
-                        bc "here"
+              afterDamage {
+                apply ASLEEP
+                def pcs = defending
+                targeted(pcs) {
+                  if (pcs.isSPC(ASLEEP)) {// Is !bg.em().run(new ApplySpecialCondition(POISONED, pcs, SOURCE.ATTACK)) better here
+                    delayed {
+                      before ASLEEP_SPC, null, null, BEGIN_TURN, {
+                        flip "Asleep (Endless Darkness)", 2, {}, {}, [2: {
+                          ef.unregisterItself(bg.em());
+                        }, 1:{
+                          bc "$ef.target is still asleep."
+                        }, 0:{
+                          bc "$pcs is knocked out by $thisMove."
+                          new Knockout(pcs).run(bg)
+                        }]
+                        prevent()
```

Comment:
This seems like it could have some weird timing issue at some point and could end up applying when it shouldn't. Should definitely try weird combinations for testing.

Reply:
I've done a lot of testing on this card and haven't been able to trigger it when it was not appropriate.